src/coreclr/nativeaot/System.Private.CoreLib/src/Internal/Runtime/CompilerHelpers/ThrowHelpers.cs

```diff
@@ -132,5 +132,10 @@ public static void ThrowArgumentOutOfRangeException()
         {
             throw new ArgumentOutOfRangeException();
         }
+
+        public static void ThrowInvalidOperationInlineArrayEqualsGetHashCode()
+        {
+            throw new InvalidOperationException(SR.InvalidOperation_InlineArrayEqualsGetHashCode);
```

Comment:
Re-reading the [guidelines](https://learn.microsoft.com/en-us/dotnet/fundamentals/runtime-libraries/system-notsupportedexception#related-exception-types) about InvalidOperationException vs. NotSupportedException, I think this should be NotSupportedException.